TESTING.adoc

```diff
@@ -0,0 +1,107 @@
+
+[[structure-of-tests]]
+=== Structure of Tests
+Feature files which define the test cases for each individual add-on are stored in the *_test/integration/features_* directory.
+The feature files follow the naming convention of `<addon-name>.feature` and begin with the tag `@<addon-name>`.
+
+Integration tests are implemented in the *_test/integration/integration_test.go_* file.
+This file imports the test suite from the Minishift repository and can be used for the import of additional step definitions.
+
+[[prerequisites]]
+== Prerequisites
+You must do the following to run the tests successfully:
+
+- Install link:https://docs.openshift.org/latest/minishift/getting-started/installing.html[Minishift and its prerequisites].
+- Install a recent link:https://golang.org/dl/[Go distribution] (>=1.8).
+- Install the link:https://github.com/golang/dep[dep package] with the following: `go get -u github.com/golang/dep/cmd/dep`
+- `git clone` this repository into *_$GOPATH/src/github.com/minishift/minishift-addons_*.
+- Get the latest `minishift` binary from https://github.com/minishift/minishift/releases/latest and place it in *_testing/bin_*.
+- Vendor the needed Go packages by runining `make vendor` in the root directory of the *minishift-addons* repository.
+
+[NOTE]
+====
+We recommend vendoring the packages after every `git pull` from the *minishift-addons* master branch.
+====
+
+[[running-tests]]
+== Running Tests
+To run integration tests for a specific add-on, go to the root directory of the *minishift-addons* repository and run the following command: 
+
+```
+make integration ADDON=<name-of-addon>
+```
+
+This command will start tests on all Gherkin feature files present in *_tests/integration/features_* which contain the tag `@<name-of-addon>`.
+For most add-ons, this will mean using one feature file located at *_tests/integration/features/<name-of-addon>.feature_*.
+
+To run the tests on all add-ons, you can use the following:
+
+```
+make integration
+```
+
+[[additional-options]]
+=== Additional Options
+It is possible to run the tests with a Minishift binary located outside of the default location of *_testing/bin_*.
+To do so, use the `MINISHIFT_BINARY` parameter with the `make integration` target.
+For example:
+
+```
+make integration ADDON=xpaas MINISHIFT_BINARY=/home/user/minishift/minishift
+```
+
+[[writing-tests]]
+== Writing Tests
+Create a new file named *_tests/integration/features/<name-of-add-on>.feature_* to create a new integration test for an add-on.
+The *_example.feature_* file demonstrates the basic structure of an integration test for an example add-on and may be used as a starting point.
+
+Features should contain at least 6 key parts which secure that the tested add-on can be installed, applied, removed and uninstalled:
```

Comment:
@jorgemoralespou Could you look into this documentation on writing tests. Also see `Advanced steps`. There is a link to https://github.com/minishift/minishift/tree/master/test/integration/testsuite for more reference.